src/components/popover/popover.ts

```diff
@@ -61,15 +61,51 @@ export class MdlPopoverComponent implements AfterViewInit {
     }
 
     private hideAllPopovers() {
-        [].map.call(
-          document.querySelectorAll('.mdl-popover.is-visible'),
-          (el: Element) => el.dispatchEvent(new Event('hide'))
-        );
+      let nodeList = document.querySelectorAll('.mdl-popover.is-visible');
+      for(let i=0; i < nodeList.length;++i) {
+        nodeList[i].dispatchEvent(new Event('hide'));
+      }
     }
 
     private show(event: Event) {
         event.stopPropagation();
         this.isVisible = true;
+        this.updateDirection(event);
+    }
+
+    private updateDirection(event: Event) {
+      // To prevent toggle test from failing, since jasmines SpyObj EventTarget didn't work as expected
+      if(!event.target) {
+        return;
+      }
+      const nativeEl = this.elementRef.nativeElement;
+      const targetRect = (<HTMLElement>event.target).getBoundingClientRect();
+      const viewHeight = window.innerHeight;
+
+      // In case something goes wrong, pollBuffer is there to stop avoid memory leak
+      let pollBuffer = 0;
+      let longPoll = setInterval(() => {
```

Comment:
@nikoTM I see, I was not aware it was 400ms. Thanks for pointing it :)

Any form of waiting for dropdown to expand to measure it and decided on direction creates flickering. I was not able to see difference between the setTimeout and setInterval. They both have flickering on first click when direction changes.

This can be fixed by including the node dimension code as helper here is commit https://github.com/tb/angular2-mdl-ext/commit/8856d5c94e59aa7f9966fdad4ef9896a57652f36 
@mseemann What do you think of including such helper code? Where should it be located?